advanced_alchemy/types.py

```diff
@@ -174,3 +175,42 @@ def process_result_value(self, value: datetime.datetime | None, dialect: Dialect
 """A JSON type that uses  native ``JSONB`` where possible and ``Binary`` or ``Blob`` as
 an alternative.
 """
+
+
+class EncryptedString(TypeDecorator):
+    """Encrypted String
+
+    Configurable encrypted string field for SQL Alchemy
+
+    Raises:
+        NotImplementedError: Unsupported backend
+    """
+
+    # todo: should be BYTEA for postgres.  determine best field for others.  # noqa: FIX002
+    impl = BINARY
+
+    cache_ok = True
+
+    def __init__(self, passphrase: str, backend: Literal["pgcrypto", "tink", "cryptography"] = "cryptography") -> None:
+        super().__init__()
+        self.passphrase = passphrase
+        self.backend = backend
+
+    def bind_expression(self, bindparam: Any) -> Any:
+        # convert the bind's type from EncryptedString to
+        # String, so that it's passed as is without
+        # a dbapi.Binary wrapper
+        bindparam = type_coerce(bindparam, String)
+        return self._handle_encrypted_data(bindparam)
+
+    def column_expression(self, column: Any) -> Any:
+        return self._handle_encrypted_data(column)
+
+    def _handle_encrypted_data(self, column: Any) -> str:
+        if self.backend == "pgcrypto":
+            return sql_func.pgp_sym_encrypt(column, self.passphrase)  # type: ignore[return-value]
+        if self.backend == "tink":
+            raise NotImplementedError
+        if self.backend == "cryptography":
+            raise NotImplementedError
```

Comment:
Wouldn't it make more sense to not have these in the first place if they're currently not implemented?

Sounds less confusing to me.

Reply:
I expect to have them before this PR is merged.  I was really trying to set up the structure so I could think through the whole thing.

Reply:
Oh. Sorry for the slightly overzealous comments then :slightly_smiling_face: